content/api/snippets.apib

```diff
@@ -0,0 +1,198 @@
+        + Default: false
+
+
++ Request (application/json)
+
+    + Body
+
+        ```
+        {
+            "id" : "header",
+            "content": {
+                "html": "<b>Our cool header snippet for {{name}}</b>"
+            }
+        }
+        ```
+
++ Response 200 (application/json)
+
+        {
+          "results": {
+            "id": "header"
+          }
+        }
+
++ Response 422 (application/json)
+
+        {
+          "errors" : [
+            {
+              "part" : "text",
+              "description" : "Error while compiling part text: line 4: syntax error near 'age'",
+              "line" : 4,
+              "code" : "3000",
+              "message" : "substitution language syntax error in snippet content"
+            }
+          ]
+        }
+
+## Retrieve a Snippet [GET /snippets/{id}]
+
++ Parameters
+    + id (required, string, `ourfooter`)
+
++ Request
+
+
++ Response 200 (application/json)
+
+        {
+          "results" : {
+            "id" : "ourfooter",
+            "content": {
+              "html" : "<b>Our standard footer</b>",
+              "text" : "Our standard footer",
+              "amp_html" : "<b>Our standard footer</b>"
+            }
+          }
+        }
+
+## Update a Snippet [PUT /snippets/{id}]
+
+If a content object is provided in the update request, it must
+contain all relevant content fields whether they are being changed or not.
+The new content will completely overwrite the existing content.
+
++ Data Structure
+    + content (object) - HTML, AMPHTML, and/or text content. Substitution syntax is supported with the exception of the render_dynamic_content() and render_snippet() macro calls. In other words, snippets may not reference other snippets.
+        + html (string) - HTML content which will be used when the snippet is referenced from the HTML part of a template or inline content. Maximum length - 100KB.
+        + text (string) - Plain text content which will be used when the snippet is referenced from the plain text part of a template or inline content. Maximum length - 100KB.
+        + amp_html (string) - AMPHTML content which will be used when the snippet is referenced from the AMPHTML part of a template or inline content. Maximum length - 100KB.
+    + shared_with_subaccounts (boolean) - Whether this snippet can be used by subaccounts.
+
+
++ Parameters
+    + id (required, string, `ourfooter`)
+
++ Request Update (application/json)
+
+    + Body
+
+        ```
+       {
+         "content": {
+           "html" : "<b>Our updated footer</b>",
+           "text" : "Our updated footer",
+           "amp_html" : "<b>Our updated footer</b>"
+          }
+        }
+        ```
+
++ Response 200
+
+## Delete a Snippet [DELETE /snippets/{id}]
+
+
++ Parameters
+    + id (required, string, `ourfooter`)
+
++ Request
+
++ Response 204 (application/json)
+
++ Response 404 (application/json)
+
+    + Body
+
+        ```
+        {
+        "errors": [
```

Comment:
I'd remove this from the docs.  404s are obfuscated and we're trying to remove extended errors codes